src/libstd/sys/windows/fs2.rs

```diff
@@ -111,8 +120,31 @@ impl DirEntry {
     }
 
     pub fn path(&self) -> PathBuf {
+        self.root.join(&self.file_name())
+    }
+
+    pub fn file_name(&self) -> OsString {
```

Comment:
This matches the `file_name` methods of `Path`, which was the source for this name. The directory/file distinction here is known to not apply much.